util/gnmi.go

```diff
@@ -53,12 +55,43 @@ func PathElemsEqual(a, b *gpb.PathElem) bool {
 	if a.Name != b.Name {
 		return false
 	}
+
 	if len(a.Key) != len(b.Key) {
 		return false
 	}
 
 	for k, v := range a.Key {
-		if vo, ok := b.Key[k]; !ok || vo != v {
+		if vo, ok := b.Key[k]; !ok || v != vo {
+			return false
+		}
+	}
+	return true
+}
+
+// PathElemsEqualWildcard checks if two PathElem are the same allowing for wildcard keys.
+// If a.Key["foo"] == "*" and a.Key["foo"] == "bar" func returns true.
+// If a has no keys, it is to equal to b regardless of what keys b has and vice versa.
```

Comment:
We probably should add something to the spec that defines this, but my expectation would be that `null` and `*` are the same thing - so that we say, for the list:

```yang
list foo { 
  key "one two";
}
```

* `/foo[one=*][two=*]`
* `/foo[one=*]`
* `/foo[two=*]`
* `/foo`

all mean the same thing (essentially, all entries in `foo`).

I didn't quite understand Wen's comment of "they are not on exactly the same keys" -- Wen, could you expand here?

Dan -- keys in YANG are mandatory to be specified, does your "unset key" here refer to in the supplied user wildcarded path?

Reply:
It's irrelevant now. I believe I was saying how if the ordered pair of ("*", null, ...) between the two inputs don't match exactly, e.g. ("*", null) and (null, "*"), the current function would fail. But if we agree that `null` and `*` are the same thing, then they should be equal.

Reply:
thanks for clarification, i was talking about the user supplied path